core/tests/behavior/blocking_list.rs

```diff
@@ -20,62 +20,27 @@ use std::collections::HashSet;
 
 use anyhow::Result;
 use log::debug;
-use opendal::BlockingOperator;
-use opendal::EntryMode;
-
-use super::utils::*;
-
-/// Test services that meet the following capability:
-///
-/// - can_read
-/// - can_write
-/// - can_blocking
-/// - can_list or can_scan
-macro_rules! behavior_blocking_list_test {
-    ($service:ident, $($(#[$meta:meta])* $test:ident),*,) => {
-        paste::item! {
-            $(
-                #[test]
-                $(
-                    #[$meta]
-                )*
-                fn [<blocking_list_ $test >]() -> anyhow::Result<()> {
-                    match OPERATOR.as_ref() {
-                        Some(op) if op.info().can_read()
-                            && op.info().can_write()
-                            && op.info().can_blocking() && op.info().can_list() => $crate::blocking_list::$test(op.blocking()),
-                        Some(_) => {
-                            log::warn!("service {} doesn't support blocking_list, ignored", opendal::Scheme::$service);
-                            Ok(())
-                        },
-                        None => {
-                            Ok(())
-                        }
-                    }
-                }
-            )*
-        }
-    };
-}
 
-#[macro_export]
-macro_rules! behavior_blocking_list_tests {
-     ($($service:ident),*) => {
-        $(
-            behavior_blocking_list_test!(
-                $service,
-
-                test_list_dir,
-                test_list_non_exist_dir,
-                test_scan,
-                test_remove_all,
-            );
-        )*
-    };
+use crate::*;
+
+pub fn behavior_blocking_list_tests(op: &Operator) -> Vec<Trial> {
+    let cap = op.info().capability();
+
+    if !(cap.read && cap.write && cap.copy && cap.blocking && cap.list) {
+        return vec![];
+    }
+
+    blocking_trials!(
+        op,
+        test_blocking_list_dir,
+        test_blocking_list_non_exist_dir,
+        test_blocking_scan,
+        test_remove_all
```

Comment:
We should rename `test_remove_all` to `test_blocking_remove_all` for consistentcy.

Reply:
Fixed.